contractsdk/cpp/xchain/basic_iterator.cc

```diff
@@ -0,0 +1,78 @@
+#include "xchain/xchain.h"
+#include "xchain/basic_iterator.h"
+#include "xchain/syscall.h"
+
+namespace xchain {
+
+Iterator::Iterator(const std::string& start, const std::string& limit, size_t cap) {
+    _cap = cap;
+    _start = start;
+    _limit = limit;
+
+    if (!load()) {
+        _it = -1;
+    } else {
+        _it = 0;
+    }
+}
+
+bool Iterator::load() {
+    _buf.clear();
+    //每次多取一个作为下一次的主键
+    bool ok = range_query(_start, _limit, _cap + 1, &_buf);
+    if (!ok) {
+        error = Error(xchain::ErrorType::kErrIteratorLoad);
+        return false;
+    }
+    if (_buf.size() != 0) {
+        if (_buf.size() == _cap + 1) {
+            _start = (*(_buf.rbegin())).first;
+            _buf.pop_back();
+        } else {
+            _start = _limit;
+        }
+    }
+    return true;
+}
+
+bool Iterator::next() {
+    bool ret = end();
```

Comment:
if the value of ret equals true, it means that _it is out of bounds. So next statement that getting the value belonged to index _it, is it illegal?